subsys/bluetooth/mesh/shell/shell_dtt_cli.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "shell_utils.h"
+
+static struct bt_mesh_model *mod;
+
+static void dtt_print(const struct shell *shell, int err, int32_t rsp)
+{
+	if (!err) {
+		shell_print(shell, "Transition time: %d", rsp);
+	}
+}
+
+static int shell_dtt_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_GEN_DEF_TRANS_TIME_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_dtt_cli *cli = mod->user_data;
+	int32_t rsp;
+
+	int err = bt_mesh_dtt_get(cli, NULL, &rsp);
+
+	dtt_print(shell, err, rsp);
+	return err;
+}
+
+static int dtt_set(const struct shell *shell, size_t argc, char *argv[], bool acked)
+{
+	int32_t trans_time = (int32_t)strtol(argv[1], NULL, 0);
+
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_GEN_DEF_TRANS_TIME_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_dtt_cli *cli = mod->user_data;
+
+	if (acked) {
+		int32_t rsp;
+		int err = bt_mesh_dtt_set(cli, NULL, trans_time, &rsp);
+
+		dtt_print(shell, err, rsp);
+		return err;
+	} else {
+		return bt_mesh_dtt_set_unack(cli, NULL, trans_time);
+	}
+}
+
+static int shell_dtt_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	return dtt_set(shell, argc, argv, true);
+}
+
+static int shell_dtt_set_unack(const struct shell *shell, size_t argc, char *argv[])
+{
+	return dtt_set(shell, argc, argv, false);
+}
+
+static int shell_instance_get_all(const struct shell *shell, size_t argc, char *argv[])
+{
+	return shell_model_instances_get_all(shell, BT_MESH_MODEL_ID_GEN_DEF_TRANS_TIME_CLI);
+}
+
+static int shell_instance_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	uint8_t elem_idx = (uint8_t)strtol(argv[1], NULL, 0);
+
+	return shell_model_instance_set(shell, &mod, BT_MESH_MODEL_ID_GEN_DEF_TRANS_TIME_CLI,
+					elem_idx);
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(instance_cmds,
+			       SHELL_CMD_ARG(set, NULL, "<elem_idx> ", shell_instance_set, 2, 0),
+			       SHELL_CMD_ARG(get-all, NULL, NULL, shell_instance_get_all, 1, 0),
+			       SHELL_SUBCMD_SET_END);
+
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	dtt_cmds, SHELL_CMD_ARG(get, NULL, NULL, shell_dtt_get, 1, 0),
+	SHELL_CMD_ARG(set, NULL, "<transition_time_ms>", shell_dtt_set, 2, 0),
```

Comment:
Could we avoid using `shell_` prefix in internal functions? They may confuse with shell API. Especially these `shell_instance_*` functions. I think `cmd_` prefix is better. And also aligned with the upstream `mesh/shell.c`.

P.S. This is the last nit picking!